### PR TITLE
Pd 446 document storage snapshot namespace

### PR DESCRIPTION
**Pull request by micjohnson777**
This PR updates the CLIESnapshots.md article with command content and standard namespace content.
It merges into the 23.10 branch.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

File: content/SCALECLIReference/Storage/CLISnapshot.md [hunk 1]
```diff
@@ -14,5 +14,385 @@ tags:
...
+## Snapshot Namespace
+The **snapshot** namespace has 10 command(s), and is based on snapshot creation and management functions found in the SCALE API and web UI.
+It provides access to storage snapshot methods through the **snapshot** commands.
+
+## Snapshot Commands 
+The following **snapshot** commands allow you to create new snapshots, and manage existing snapshots.
+
+You can enter commands from the main CLI prompt or from the snapshot namespace prompt.
+
+{{< include file="/_includes/CLI/CLIGuideWIP.md" type="page" >}}
```
Comment (DjP-iX) [inline content/SCALECLIReference/Storage/CLISnapshot.md:26]:
```suggestion
```

File: content/SCALECLIReference/Storage/CLISnapshot.md [hunk 2]
```diff
@@ -14,5 +14,385 @@ tags:
...
+{{< expand "Command Example" "v" >}}
+```
+storage snapshot clone snapshot_remove={["snapshot"="tank/minio@auto-2023-08-16_00-00"], ["dataset_dst"="tank2/snapshots"]}
+true
+```
+{{< /expand >}}
+{{< /expand >}} -->
+
+### Rollback Command  
+The `rollback` command replaces data in the specified dataset with the information save in the snapshot matching the ID entered.
```
Comment (DjP-iX) [inline content/SCALECLIReference/Storage/CLISnapshot.md:332]:
```suggestion
The `rollback` command replaces data in the specified dataset with the information saved in the snapshot matching the ID entered. 
```

**Review by DjP-iX — CHANGES_REQUESTED** [review]

File: content/SCALECLIReference/Storage/CLISnapshot.md [hunk 3]
```diff
@@ -14,5 +14,385 @@ tags:
...
+{{< expand "Using the Create Command" "v" >}} 
+#### Description  
+The `create` has one required property, `dataset`, and seven optional properties.
+The **Create Command Optional Properties** section below lists the seven optional properties and provides syntax examples. 
+`dataset` defines the pool/dataset path for the snapshot you want to create. For example, *tank/minio*.
+Enter a property argument using the `=` delimiter to separate property and value. Enter string values in double quotes. 
+<!-- comment out this until array syntax is verified
+If entering a property argument with multiple values, enclose the double-quoted property and values in square brackets `[]`, and separate each with a comma and space.
+Enclose array properties arguments in curly brackets `{}`. 
+Enclose each property arguments in the array in `[]`square brackets, with double-quoted property and values and using `=` delimiter to separate them.
```
[the remark this answers is not in the document]
Comment (micjohnson777) [reply content/SCALECLIReference/Storage/CLISnapshot.md:85]: agree. I just left it for now.

File: content/SCALECLIReference/Storage/CLISnapshot.md [hunk 4]
```diff
@@ -14,5 +14,385 @@ tags:
...
+```
+storage snapshot create dataset="tank/minio" name=miniosnaps
+
+```
+{{< /expand >}}
+{{< /expand >}}
+
+### Delete Command 
+The `delete` command deletes a snapshot matching the ID entered. 
+Use the <code>[query](#query-command)</code> command to locate the snapshot ID.
```
[the remark this answers is not in the document]
Comment (micjohnson777) [reply content/SCALECLIReference/Storage/CLISnapshot.md:138]: cool... I didn't know that!

File: content/SCALECLIReference/Storage/CLISnapshot.md [hunk 5]
```diff
@@ -14,5 +14,385 @@ tags:
...
+
+### Remove Command  
+The `remove` command removes a snapshot from a given dataset.
+Use the <code>[query](#query-command)</code> command to locate the snapshot ID.
+
+{{< include file="/_includes/CLI/CLICommandWIP.md" type="page" >}}
+<!-- cannot get the command to work, commenting out until syntax verified and that command works
+{{< expand "Using the Remove Command" "v" >}} 
+#### Description  
+`remove` command has one required property, `snapshot_remove`.
```
[the remark this answers is not in the document]
Comment (micjohnson777) [reply content/SCALECLIReference/Storage/CLISnapshot.md:294]: I missed that on. oops.

**Review by DjP-iX — APPROVED** [review]

**Pull request merged**